src/Symfony/Component/VarDumper/Server/Connection.php

```diff
@@ -47,66 +45,52 @@ public function getContextProviders(): array
         return $this->contextProviders;
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function dump(Data $data, $output = null): void
+    public function write(Data $data): bool
     {
         set_error_handler(array(self::class, 'nullErrorHandler'));
-
-        $failed = false;
         try {
             if (!$this->socket = $this->socket ?: $this->createSocket()) {
-                $failed = true;
-
-                return;
+                return false;
             }
         } finally {
             restore_error_handler();
-            if ($failed && $this->wrappedDumper) {
-                $this->wrappedDumper->dump($data);
-            }
         }
 
-        set_error_handler(array(self::class, 'nullErrorHandler'));
-
-        $context = array('timestamp' => time());
+        $context = array('timestamp' => microtime(true));
         foreach ($this->contextProviders as $name => $provider) {
             $context[$name] = $provider->getContext();
         }
         $context = array_filter($context);
-
         $encodedPayload = base64_encode(serialize(array($data, $context)))."\n";
-        $failed = false;
 
+        set_error_handler(array(self::class, 'nullErrorHandler'));
         try {
             $retry = 3;
-            while ($retry > 0 && $failed = (-1 === stream_socket_sendto($this->socket, $encodedPayload))) {
+            while (true) {
+                if (-1 !== stream_socket_sendto($this->socket, $encodedPayload)) {
+                    return true;
+                }
                 stream_socket_shutdown($this->socket, STREAM_SHUT_RDWR);
-                if ($failed = !$this->socket = $this->createSocket()) {
+
+                if (0 === --$retry || !$this->socket = $this->createSocket()) {
                     break;
                 }
-
-                --$retry;
             }
         } finally {
             restore_error_handler();
-            if ($failed && $this->wrappedDumper) {
-                $this->wrappedDumper->dump($data);
-            }
         }
+
+        return false;
     }
 
     private static function nullErrorHandler()
     {
-        // noop
+        // no-op
     }
 
     private function createSocket()
     {
-        $socket = stream_socket_client($this->host, $errno, $errstr, 1, STREAM_CLIENT_CONNECT | STREAM_CLIENT_PERSISTENT);
-
-        if ($socket) {
+        if ($socket = stream_socket_client($this->host, $errno, $errstr, 0, STREAM_CLIENT_CONNECT | STREAM_CLIENT_ASYNC_CONNECT | STREAM_CLIENT_PERSISTENT)) {
```

Comment:
the equivalent monolog code connects in async mode, not sure why that's not the case here?